### PR TITLE
AP-4762 Symbols detection

### PR DESCRIPTION
**Pull request by CarlosGamero**
## Changes

isTextTranslatable consider symbols and numbers as non translatable

## Checklist

- [x] Apply one of following labels; `major`, `minor`, `patch` or `skip-release`
- [x] I've updated the documentation, or no changes were necessary
- [x] I've updated the tests, or no changes were necessary

### CODE REVIEW

File: packages/app/non-translatable-markup/src/methods/index.ts [hunk 1]
```diff
@@ -1,2 +0,0 @@
-export { isAttemptToEditNonTranslatableContent } from './isAttemptToEditNonTranslatableContent'
```
Comment (CarlosGamero) [inline packages/app/non-translatable-markup/src/methods/index.ts:1]: Removing this file as discussed on prev PR

File: packages/app/non-translatable-markup/src/methods/isAttemptToEditNonTranslatableContent.ts [hunk 2]
```diff
@@ -9,9 +8,8 @@ export const isAttemptToEditNonTranslatableContent = (
   text: string,
   updatedText: string,
 ): boolean => {
-  if (!isTextTranslatable(updatedText)) {
-    return true
-  }
+  // early return if the text is the same
+  if (text === updatedText) return false
```
Comment (CarlosGamero) [inline packages/app/non-translatable-markup/src/methods/isAttemptToEditNonTranslatableContent.ts:12]: `isTextTranslatable` can't be used here now, but looking at the method I think that it is fine to stay without it and added a new condition which is a better early return

**Comment by CarlosGamero** [comment]
Lint issue addressed on -> https://github.com/lokalise/shared-ts-libs/pull/266

File: packages/app/non-translatable-markup/src/methods/utils.ts [hunk 3]
```diff
@@ -1,8 +1,28 @@
...
+ *      `\p{Emoji}` -> matches any emoji character
+ *  `u` -> enables unicode mode
+ *
+ * Note: `\p{S}` matches any kind of symbol character, and it covers all the symbols from `\p{Sm}`, `\p{Sc}`, `\p{Sk}`,
+ *  and `\p{So}` categories. However, we don't want to cover `\p{Sc}` (Currency Symbol), so we specify the relevant
+ *  subcategories separately.
+ *
+ * For more info see -> https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/regular_expressions/unicode_character_class_escape/index.md
+ */
+const symbolsPattern = /[\p{P}\p{Sm}\p{Sk}\p{So}\p{Emoji}']/u
```
Comment (kjamrog) [inline packages/app/non-translatable-markup/src/methods/utils.ts:26]: If I see correctly, for checking symbols you split text by symbols and then check each part recursively? Won't it work if we use:
```
/[\p{P}\p{Sm}\p{Sk}\p{So}\p{Emoji}']+/u
```
And then check if whole string is matching regex?

File: packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts [hunk 4]
```diff
@@ -1,10 +1,25 @@
...
-  const parts = text.split(nonTranslatableTextRegexp).map((part) => part.trim())
+export const isTextTranslatable = (text: string): boolean =>
+  trimAndFilterEmpty(text.split(nonTranslatableTextRegexp)).some((piece) =>
+    isPieceTranslatable(piece),
+  )
 
-  return parts.some((part) => part !== '')
+const isPieceTranslatable = (piece: string): boolean => {
+  // numbers are not translatable
+  if (!Number.isNaN(+piece)) return false
```
Comment (kjamrog) [inline packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts:15]: Maybe for simplicity we could include numbers in regex pattern?

Comment (CarlosGamero) [reply packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts]: Good point! done, thank youuu

File: packages/app/non-translatable-markup/src/methods/utils.ts (hunk 3 above)
Comment (CarlosGamero) [reply packages/app/non-translatable-markup/src/methods/utils.ts]: Fixed, thank youuu

**Review by kjamrog — APPROVED** [review]

File: packages/app/non-translatable-markup/src/methods/isAttemptToEditNonTranslatableContent.ts (hunk 2 above)
Comment (kibertoad) [reply packages/app/non-translatable-markup/src/methods/isAttemptToEditNonTranslatableContent.ts:12]: is the inversion from false to true intended here?

File: packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts [hunk 5]
```diff
@@ -1,10 +1,16 @@
...
+ * Returns true if the text contain at least one translatable piece of text.
+ * A translatable piece of text is a piece of text that is not surrounded by non-translatable tags,
+ * symbols or a number.
  */
-export const isTextTranslatable = (text: string): boolean => {
-  const parts = text.split(nonTranslatableTextRegexp).map((part) => part.trim())
+export const isTextTranslatable = (text: string): boolean =>
+  text
+    .split(nonTranslatableTextRegexp)
+    .map((p) => p.trim())
```
Comment (kibertoad) [inline packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts:11]: it's not clear what `p` stands for, can we make it more explicit?

File: packages/app/non-translatable-markup/src/methods/isAttemptToEditNonTranslatableContent.ts (hunk 2 above)
Comment (CarlosGamero) [reply packages/app/non-translatable-markup/src/methods/isAttemptToEditNonTranslatableContent.ts:12]: Yep, is `text` and `updatedText` is the same it means that user is not trying to edit non translatable content

File: packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts (hunk 5 above)
Comment (CarlosGamero) [reply packages/app/non-translatable-markup/src/methods/isTextTranslatable.ts]: Good point! already done, thank youuu

**Review by kibertoad — APPROVED** [review]

**Pull request merged**